File: CSharpSrc/Token.cs
namespace ConfigureParser
{
    internal enum TokenType
    {
        Word,
        LeftBracket,
        RightBracket,
        LeftSquareBracket,
        RightSquareBracket,
    }

    internal class Token
    {
        public string Content { get; }
        public int Level { get; }
        public TokenType Type { get; }

        public Token(string content, int level, TokenType type)
        {
            Content = content;
            Level = level;
            Type = type;
        }
    }
}

File: JavaSrc/com/company/Token.java
package com.company;

/**
 * Created by LaoFan on 3/30/19.
 */
enum TokenType {
    word,
    leftBracket,
    rightBracket,
    leftSquareBracket,
    rightSquareBracket,
}

public class Token {
    private String content;
    private int level;
    private TokenType type;

    public Token(String content, int level, TokenType type) {
        this.content = content;
        this.level = level;
        this.type = type;
    }

    public String content() {
        return content;
    }

    public int level() {
        return level;
    }

    public TokenType type() {
        return type;
    }
}
